tests/callbacks/test_callback_hook_outputs.py

```diff
@@ -71,3 +71,45 @@ def on_train_epoch_end(self, outputs) -> None:
 
     results = trainer.fit(model)
     assert results
+
+
+def test_on_val_epoch_end_outputs(tmpdir):
+
+    class CB(Callback):
+
+        def on_validation_epoch_end(self, trainer, pl_module, outputs):
+            if trainer.running_sanity_check:
+                assert len(outputs[0]) == trainer.num_sanity_val_batches[0]
+            else:
+                assert len(outputs[0]) == trainer.num_val_batches[0]
+
+    model = BoringModel()
+
+    trainer = Trainer(
+        callbacks=CB(),
+        default_root_dir=tmpdir,
+        limit_train_batches=2,
+        limit_val_batches=2,
+        max_epochs=1,
```

Comment:
```suggestion
        fast_dev_run=2,
```

Reply:
Need to check for `snaity_check` as well. `fast_dev_run` would skip that.